pyro/distributions/lkj.py

```diff
@@ -202,9 +202,11 @@ def log_prob(self, x):
         Km1 = self._d - 1
 
         log_diagonals = x.diagonal(offset=0, dim1=-1, dim2=-2)[..., 1:].log()
+        # TODO: Figure out why the `device` kwarg to torch.linspace seems to not work in certain situations,
+        # and a seemingly redundant .to(x.device) is needed below.
```

Comment:
shouldnt we just remove the `device=x.device` then?

Reply:
The device arg to `torch.linspace` should work fine, and I'm only facing this issue when I run `test_cuda.py`. Having the `.to` should be redundant in most cases, but removing the `device` arg will be 2-3 times slower.